resources/js/Components/PortfolioBreakdown/Hooks/usePortfolioBreakdown.ts

```diff
@@ -1,32 +1,37 @@
+import { type QueryKey, useQuery, useQueryClient } from "@tanstack/react-query";
 import axios from "axios";
-import { useState } from "react";
 import { GRAPH_BACKGROUND_COLORS } from "@/Components/PortfolioBreakdown/PortfolioBreakdown.contracts";
+import { useWalletPollingInterval } from "@/Hooks/useWalletPollingIntervals";
+import { isTruthy } from "@/Utils/is-truthy";
 
-export const usePortfolioBreakdown = (): {
+export const usePortfolioBreakdown = (
+    wallet?: App.Data.Wallet.WalletData | null,
+): {
     assets: App.Data.TokenPortfolioData[];
-    loading: boolean;
-    loadBreakdown: () => Promise<void>;
+    isLoading: boolean;
 } => {
-    const [loading, setLoading] = useState<boolean>(true);
-    const [assets, setAssets] = useState<App.Data.TokenPortfolioData[]>([]);
+    const { calculateInterval } = useWalletPollingInterval(wallet);
 
-    const loadBreakdown = async (): Promise<void> => {
-        setLoading(true);
+    const queryClient = useQueryClient();
+    const queryKey: QueryKey = ["breakdown"];
 
-        const { data } = await axios.get<App.Data.TokenPortfolioData[]>(
-            route("tokens.breakdown", {
-                top_count: GRAPH_BACKGROUND_COLORS.length - 1,
-            }),
-        );
-
-        setAssets(data);
-
-        setLoading(false);
-    };
+    const { data, isLoading } = useQuery({
+        enabled: isTruthy(wallet),
+        queryKey,
+        refetchInterval: () => calculateInterval(queryClient.getQueryState(queryKey)?.dataUpdateCount),
```

Comment:
Am I missing something or are we potentially passing `null/undefined` for `CalculateIntervalProperties`? 

The signature here https://github.com/ArdentHQ/dashbrd/pull/24/files/13a52a45da97af5e329b7292f2178bdb89c40a30#diff-31a1e71a0d82b031fc5bbf8ff7bb1ac29de33f4dc2ccfbd7a9b5896fa7792fefR21 should be changed to reflect it.

Also double check if `dataUpdateCount` is actually an object.